app/models/test_track/identity_session_locator.rb

```diff
@@ -8,8 +8,8 @@ def initialize(identity)
   def with_visitor
     raise ArgumentError, "must provide block to `with_visitor`" unless block_given?
 
-    if matching_identity?
-      yield session.visitor_dsl
+    if web_context?
+      yield session.visitor_dsl_for(identity)
```

Comment:
Eventually I think we'll require that a session be present whenever TestTrack splits are exercised and this whole web context thing will go away, and it'll naturally be wired into job lifecycles, etc. But for now, the key change here is that if you've got a TT session (you're in a "web context"), we're going to expect the online session to cough up a TT visitor for us - even if it's not the same identity that is logged into the web.